lib/sitemap/sitemap_generator.rb

```diff
@@ -34,15 +35,26 @@ def sitemaps
   def generate_xml(chunk)
     builder = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
       xml.urlset(xmlns: "http://www.sitemaps.org/schemas/sitemap/0.9") do
-        chunk.each do |url|
-          url = Array(url).first
-          url = URI.join(base_url, url) unless url.start_with?("http")
-          xml.url {
-            xml.loc url
-          }
+        chunk.each do |document|
+          if document == "/"
```

Comment:
👍 I've added a struct which responds to `link` and `public_timestamp`. What do you think?